docs/design-patterns.mdx

```diff
@@ -74,75 +180,279 @@ and in the transaction that is used to deploy it:
-        }
     }
 }
-...
-import A from 0xA
 
-pub fun main(): A.BReport {
-    let b: AContract.BResource // Borrow the resource
-    let report = b.generateReport()
-    return b
+// Uses borrow instead to avoid costly operations
+// GOOD
+transaction {
+
+    prepare(acct: AuthAccount) {
+
+        // Borrows a reference to the stored vault, much less costly operation
+        let vault <- acct.borrow<&ExampleToken.Vault>(from: /storage/exampleToken)
+
+        let burnVault <- vault.withdraw(amount: 10)
+
+        destroy burnVault
+
+        // No `save` required because we only used a reference
+    }
 }
 ```
+
+# Capabilities
+
+## Capability Receiver
+
+### Problem
+
+An account must be given a [capability](/cadence/language/capability-based-access-control/)
+to a resource or contract in another account but a single transaction with both accounts authorizing it
+cannot be easily produced. This prevents a single transaction from fetching the capability
+from one account and delivering it to the other.
+
+### Solution
+
+Account B creates a resource that can receive the capability and stores this in their `/storage/` area.
+They then expose a Capability to this in their `/public/` area with a function
+that can receive the desired Capability and store it in the resource for later use.
+
+Account A fetches the receiver Capability from B's `/public/` area, creates the desired Capability,
+and passes it to the receiver function. The receiver function stores the Capability
+in the resource that it is on in account B's `/storage/` area for later use.
+
+There are two nuances to this workflow that are required to ensure that it is secure.
+
+The first is that only Account A should be able to create instances of the desired Capability.
+This ensures that nobody else can create instances of it and call the receiver function
+on B's receiver capability instead of A. This means that A is probably an admin account.
+
+The second is that the field on the receiver resource that stores the desired Capability
+should be `access(contract)` and only accessed by code within the contract that needs to.
+This ensures that B cannot copy and share the Capability with anyone else.
+
+### Example Code
+
+See:
+
+[LockedTokens.cdc](https://github.com/onflow/flow-core-contracts/blob/bfb115869bd9f815cde1fe64ab6d91ca95c0938e/contracts/LockedTokens.cdc#L527-L546)
+
+[custody_setup_account_creator.cdc](https://github.com/onflow/flow-core-contracts/blob/79941fe65b634800065a440ae5243744b2ca8a2f/transactions/lockedTokens/admin/custody_setup_account_creator.cdc)
+
+[admin_deposit_account_creator.cdc](https://github.com/onflow/flow-core-contracts/blob/79941fe65b634800065a440ae5243744b2ca8a2f/transactions/lockedTokens/admin/admin_deposit_account_creator.cdc)
+
+## Capability Revocation
+
+### Problem
+
+A capability provided by one account to a second account must able to be revoked
+by the first account without the co-operation of the second.
+
+### Solution
```

Comment:
Maybe add a callout and link to the [Capability Controller FLIP](https://github.com/onflow/flow/pull/798), which aims to improve this